mico-core/src/main/java/io/github/ust/mico/core/resource/ApplicationResource.java

```diff
@@ -208,7 +209,29 @@
         MicoServiceDeploymentInfo serviceDeploymentInfo;
         try {
             serviceDeploymentInfo = broker.addMicoServiceToMicoApplicationByShortNameAndVersion(
-                applicationShortName, applicationVersion, serviceShortName, serviceVersion);
+                applicationShortName, applicationVersion, serviceShortName, serviceVersion, Optional.empty());
+        } catch (MicoApplicationNotFoundException | MicoServiceNotFoundException | MicoServiceDeploymentInformationNotFoundException | KubernetesResourceException e) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, e.getMessage());
+        } catch (MicoServiceAddedMoreThanOnceToMicoApplicationException | MicoApplicationIsNotUndeployedException | MicoApplicationDoesNotIncludeMicoServiceException e) {
+            throw new ResponseStatusException(HttpStatus.CONFLICT, e.getMessage());
+        } catch (MicoTopicRoleUsedMultipleTimesException | KafkaFaasConnectorNotAllowedHereException e) {
+            throw new ResponseStatusException(HttpStatus.UNPROCESSABLE_ENTITY, e.getMessage());
+        }
+
+        return ResponseEntity.ok(new Resource<>(new MicoServiceDeploymentInfoResponseDTO(serviceDeploymentInfo)));
+    }
+
+    @ApiOperation(value = "Reuses an existing service deployment information instance and adds it to the application.")
```

Comment:
Comment out the api decorators so to not expose this endpoint in the actual api. Add a comment linking to the issue for when to uncomment these lines.